config/aaa.py

```diff
@@ -61,20 +71,69 @@ def fallback(option):
 authentication.add_command(fallback)
 
 
+# cmd: aaa authentication debug
+@click.command()
+@click.argument('option', type=click.Choice(["enable", "disable", "default"]))
+def debug(option):
+    """AAA debug [enable | disable | default]"""
+    if option == 'default':
+        del_table_key('AAA', 'authentication', 'debug')
+    else:
+        if option == 'enable':
+            add_table_kv('AAA', 'authentication', 'debug', True)
+        elif option == 'disable':
+            add_table_kv('AAA', 'authentication', 'debug', False)
+authentication.add_command(debug)
+
+
+# cmd: aaa authentication trace
+@click.command()
+@click.argument('option', type=click.Choice(["enable", "disable", "default"]))
+def trace(option):
+    """AAA packet trace [enable | disable | default]"""
+    if option == 'default':
+        del_table_key('AAA', 'authentication', 'trace')
+    else:
+        if option == 'enable':
+            add_table_kv('AAA', 'authentication', 'trace', True)
+        elif option == 'disable':
+            add_table_kv('AAA', 'authentication', 'trace', False)
+authentication.add_command(trace)
+
+
 @click.command()
-@click.argument('auth_protocol', nargs=-1, type=click.Choice(["tacacs+", "local", "default"]))
+@click.argument('auth_protocol', nargs=-1, type=click.Choice(["radius", "tacacs+", "local", "default"]))
 def login(auth_protocol):
-    """Switch login authentication [ {tacacs+, local} | default ]"""
+    """Switch login authentication [ {radius, tacacs+, local} | default ]"""
     if len(auth_protocol) is 0:
         click.echo('Argument "auth_protocol" is required')
         return
+    elif len(auth_protocol) > 2:
+        click.echo('Not a valid command.')
+        return
 
     if 'default' in auth_protocol:
+        if len(auth_protocol) !=1:
+            click.echo('Not a valid command')
+            return
         del_table_key('AAA', 'authentication', 'login')
     else:
         val = auth_protocol[0]
         if len(auth_protocol) == 2:
-            val += ',' + auth_protocol[1]
```

Comment:
a choice *only*:
s = sorted(l)
if s[0] == "local" and s[1] in ("radius", "tacacs+")